CHANGELOG.md

```diff
@@ -6,6 +6,7 @@
 
 ## Optimizations
 
+- Added more rules for simplifying expressions, especially around Concatenations. Also, meshes constructed from multiple domains are now memoized ([#2443](https://github.com/pybamm-team/PyBaMM/pull/2443))
```

Comment:
```suggestion
- Added more rules for simplifying expressions, especially around Concatenations. Also, meshes constructed from multiple domains are now memorized ([#2443](https://github.com/pybamm-team/PyBaMM/pull/2443))
```

Reply:
This refers to memoization https://en.wikipedia.org/wiki/Memoization, but I will change it to "cached"

Reply:
Oops, didn't know that, sorry! If it is a word, happy to keep it as it is.